eng/Versions.props

```diff
@@ -39,7 +39,7 @@
     <!-- Use the correct compiler version -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
     <!-- Dependencies from https://github.com/dotnet/roslyn -->
-    <MicrosoftNETCoreCompilersPackageVersion>4.0.0-4.final</MicrosoftNETCoreCompilersPackageVersion>
+    <MicrosoftNETCoreCompilersPackageVersion>4.2.0</MicrosoftNETCoreCompilersPackageVersion>
```

Comment:
Once https://github.com/dotnet/roslyn-analyzers/pull/6117 is merged, it will produce a conflict here. The version from https://github.com/dotnet/roslyn-analyzers/pull/6117 should be used.